src/api_routes_v4.rs

```diff
@@ -0,0 +1,392 @@
+          .route("/report/resolve", put().to(resolve_post_report))
+          .route("/report/list", get().to(list_post_reports))
+          .route("/site_metadata", get().to(get_link_metadata)),
+      )
+      // Comment
+      .service(
+        // Handle POST to /comment separately to add the comment() rate limitter
+        resource("/comment")
+          .guard(guard::Post())
+          .wrap(rate_limit.comment())
+          .route(post().to(create_comment)),
+      )
+      .service(
+        scope("/comment")
+          .route("", get().to(get_comment))
+          .route("", put().to(update_comment))
+          .route("/delete", post().to(delete_comment))
+          .route("/remove", post().to(remove_comment))
+          .route("/mark_as_read", post().to(mark_reply_as_read))
+          .route("/distinguish", post().to(distinguish_comment))
+          .route("/like", post().to(like_comment))
+          .route("/like/list", get().to(list_comment_likes))
+          .route("/save", put().to(save_comment))
+          .route("/list", get().to(list_comments))
+          .route("/report", post().to(create_comment_report))
+          .route("/report/resolve", put().to(resolve_comment_report))
+          .route("/report/list", get().to(list_comment_reports)),
+      )
+      // Private Message
+      .service(
+        scope("/private_message")
+          .route("/list", get().to(get_private_message))
+          .route("", post().to(create_private_message))
+          .route("", put().to(update_private_message))
+          .route("/delete", post().to(delete_private_message))
+          .route("/mark_as_read", post().to(mark_pm_as_read))
+          .route("/report", post().to(create_pm_report))
+          .route("/report/resolve", put().to(resolve_pm_report))
+          .route("/report/list", get().to(list_pm_reports)),
+      )
+      // User
+      .service(
+        scope("/account/auth")
+          .guard(guard::Post())
+          .wrap(rate_limit.register())
+          .route("/register", post().to(register))
+          .route("/login", post().to(login))
+          .route("/logout", post().to(logout))
+          .route("/password_reset", post().to(reset_password))
+          .route("/get_captcha", get().to(get_captcha))
+          .route("/password_change", post().to(change_password_after_reset))
+          .route("/change_password", put().to(change_password))
+          .route("/totp/generate", post().to(generate_totp_secret))
+          .route("/totp/update", post().to(update_totp))
+          .route("/verify_email", post().to(verify_email)),
+      )
+      .route("/account/settings/save", put().to(save_user_settings))
+      .service(
+        scope("/account/settings")
+          .wrap(rate_limit.import_user_settings())
+          .route("/export", get().to(export_settings))
+          .route("/import", post().to(import_settings)),
+      )
+      .service(
+        scope("/account")
+          .route("/my_user", get().to(get_my_user))
```

Comment:
This would be less confusing

```suggestion
          .route("", get().to(get_my_user))
```